### PR TITLE
chore(examples/templates/digitalocean-linux): make template more friendly

### PR DESCRIPTION
**Pull request by phorcys420**
This PR aims to make the DigitalOcean Linux template more user friendly for first-time users.

### CODE REVIEW

**Comment by matifali** [comment]
@phorcys420 Can we make the variable optional and only ask if there is no `DIGITALOCEAN_TOKEN`?

**Comment by phorcys420** [comment]
@matifali we can make it optional but i don't think we can hide it based on an env variable, i will try

**Comment by phorcys420** [comment]
Tested on dogfood via [relaxed_hamilton6](https://dev.coder.com/templates/coder/digitalocean-linux/versions/relaxed_hamilton6) (token revoked)

File: examples/templates/digitalocean-linux/README.md [hunk 1]
```diff
@@ -17,38 +17,36 @@ Provision DigitalOcean Droplets as [Coder workspaces](https://coder.com/docs/wor
...
+  - Note that this is only required for Fedora images to work.
 
 ### Authentication
 
-This template assumes that coderd is run in an environment that is authenticated
-with Digital Ocean. Obtain a [Digital Ocean Personal Access
-Token](https://cloud.digitalocean.com/account/api/tokens) and set the
-environment variable `DIGITALOCEAN_TOKEN` to the access token before starting
-coderd. For other ways to authenticate [consult the Terraform docs](https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs).
+This template assumes that `coderd` is run in an environment that is authenticated with Digital Ocean.
```
Comment (matifali) [inline examples/templates/digitalocean-linux/README.md:37]: Maybe we should say Coder provisioner.

File: examples/templates/digitalocean-linux/main.tf [hunk 2]
```diff
@@ -9,27 +9,26 @@ terraform {
   }
 }
 
-provider "coder" {
-}
+provider "coder" {}
 
-variable "step1_do_project_id" {
+variable "do_project_id" {
```
Comment (matifali) [inline examples/templates/digitalocean-linux/main.tf:14]: what are your thoughts on dropping `do_` prefix from all variable names?

Comment (phorcys420) [reply examples/templates/digitalocean-linux/main.tf:14]: I also thought about doing that but wasn't sure, we should do it.
`step1_do_project_id` would become `project_id` then?

Comment (matifali) [reply examples/templates/digitalocean-linux/main.tf:14]: yes

**Review by matifali — APPROVED** [review]
LGTM

**Pull request merged**